arch/posix/core/posix_core.c

```diff
@@ -131,7 +131,7 @@ static void posix_wait_until_allowed(int this_th_nbr)
 	while (this_th_nbr != currently_allowed_thread) {
 		pthread_cond_wait(&cond_threads, &mtx_threads);
 
-		if (threads_table &&
+		if ((threads_table != NULL) &&
```

Comment:
Should this be the other way around like: NULL != pointer, that way we avoid the risk of assignments instead of comparisons? (Though I must say I found this very silly rule since most people do understand these expression nowadays, and I don't think we support any architecture which NULL is not 0. )

Reply:
I don't think MISRA-C requires Yoda conditions

Reply:
The guideline does not give a big rationale for this rule. It just says that strong typing requires controlling expression of a if statement have a boolean type.